front/pages/api/w/[wId]/mcp/check_oauth.ts

```diff
@@ -0,0 +1,77 @@
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { fetchRemoteServerMetaDataByURL } from "@app/lib/actions/mcp_metadata";
+import { MCPOAuthRequiredError } from "@app/lib/actions/mcp_oauth_error";
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/auth_wrappers";
+import type { MCPOAuthExtraConfig } from "@app/lib/api/oauth/providers/mcp";
+import type { Authenticator } from "@app/lib/auth";
+import { apiError } from "@app/logger/withlogging";
+import type { WithAPIErrorResponse } from "@app/types";
+
+export type CheckOAuthResponseBody =
+  | {
+      oauthRequired: true;
+      extraConfig: MCPOAuthExtraConfig;
+    }
+  | {
+      oauthRequired: false;
+    };
+
+const PostQueryParamsSchema = t.type({
+  url: t.string,
+});
+
+async function handler(
```

Comment:
Would rename /check_oauth to  /oauth_parameters or something like that to more closely convey that we are actually creating a client id/secret pair. Maybe add a comment to make it clear that we should not be calling this a lot (probably heavily rate limited at the provider's end)